docs/learn/learn-staking-advanced.md

```diff
@@ -111,62 +111,75 @@ On Polkadot and Kusama, the instance of the pallet
-for the electing set, the nodes with 5 and 7 DOT will be selected while the node with 8 DOT will be
-left out.
-
-![bags list example 3](../assets/staking/bags-list-example-3.png)
-
-If one receives staking rewards and the amount of staked tokens within the stash account increases
-over time, the position within a bag changes and may also result in a change of bag. This may also
-happen if accounts within the bag bond more tokens or unbond tokens, one's account position and the
-position of other accounts in the bags list might change. These changes are not done automatically,
-requiring the nominator to submit the permissionless extrinsic `rebag` within the `voterList` pallet
-instance to update their position. This allows anyone to specify another account that is in the
-wrong bag, and place it in the correct one. The `voterList` pallet instance also comes with the
-extrinsic `putInFrontOf` which helps the node to move up in the bag. Actions like bonding/unbonding
-tokens automatically rebags the nominator node, but events like staking rewards/slashing do not. See
-the [bags-list](learn-nominator.md#bags-list) section for more information.
-
-This sorting functionality using bags is extremely important for the
+In {{ polkadot: Polkadot's :polkadot }}{{ kusama: Kusama's :kusama }} NPoS nomination intents are
+sorted in a list called [bags-list](https://github.com/paritytech/substrate/pull/9507).
+{{ kusama: The bags list example below uses DOT for explaining the concepts. :kusama }}The Bags-List
+substrate pallet is designed to be self-maintaining, with minimal effort from the blockchain, making
+it extremely scalable. This sorting functionality using bags is extremely important for the
 [long-term improvements](https://gist.github.com/kianenigma/aa835946455b9a3f167821b9d05ba376) of the
-staking/election system. The bags-list is capable of including an unlimited number of nodes, subject
-to the chain's runtime storage. In the current staking system configuration, the bags list keeps
+staking/election system.
+
+The bags list has two primary components, bags and nodes (or nominators' accounts), with bags
+containing the nodes with bonded balance within a specific range. In the figure below the 1st empty
+bag will contain nominators will take 21 - 30 DOT, the 2nd bag 11 - 20 DOT, and the 3rd bag 0-10
+DOT. The nomination intents are the nominators' accounts with bonded tokens (in this case they
+are 8) that will be put inside each of those three bags depending on the stake they have.
+
+![bags list example 0](../assets/bags-list-example-0.png)
+
+The bags list is semi-sorted, meaning that automatic sorting is only partially done. In fact, the
+nomination intents are automatically put into each bag based on the number of bonded tokens, but
+within each bag the nodes are arranged based on the time they are inserted and not by their stake.
+To sort nodes by stake within a bag there is a manual step (more on this later on). Note in the
```

Comment:
Sorting operation on-chain is computationally expensive. `putinfrontof` does not sort all the nodes in the bag. Instead, it just places the node infront of the nodes that have a lesser staked amount. Technically, we have to call `putinfrontof` on every single node to fully sort the bag.

```suggestion
In the
```